### ISSUE TITLE
[Feature Request] Show more session details of push/pulls

### ISSUE THREAD

**Issue by ikwyl6** [odeke-em/drive#18]
From https://github.com/rakyll/drive/issues/53

Would be nice to show, before the 'Proceed with changes', some overall details of the changes before committing:
1) Overall how many files are + and - at the end of the remote/source file comparison.
2) Number of files to be changed, shown by 1st level dirs under the 'root' dir
3) After user commits to proceeding, add push/pull kB or MB rate and total file size on the same line as:

112 / 1614 [======>----------------------------------------------------------------------------------------] 6.94 % 19h43m9s

**Comment by ikwyl6** [comment]
@odeke-em any thoughts on this one?

**Comment by odeke-em** [comment]
Got the counts in, just the kB/mB rate left. Probably will be in for release 6.

**Comment by ikwyl6** [comment]
Any update on the last feature?

**Comment by odeke-em** [comment]
I haven't yet gotten a mechanism for capturing the transfer rate aside from calculating it.

**Comment by ikwyl6** [comment]
Is there any news for the transfer rate?

**Comment by odeke-em** [comment]
None so far :(

**Comment by ikwyl6** [comment]
For transfer rate: 
https://github.com/mxk/go-flowrate
http://stackoverflow.com/questions/22421375/how-to-print-the-bytes-while-the-file-is-being-downloaded-golang

**Comment by odeke-em** [comment]
Thanks for that link. Actually I already wrote a package that produced byte updates see https://github.com/odeke-em/statos/blob/master/reader.go#L23-35 and https://github.com/odeke-em/statos/blob/master/writer.go#L23-35.
and even in use in drive this was incorporated about two months ago in April with commit https://github.com/odeke-em/drive/commit/b2659cfeaf3b689223aee3da1e26d75db7d874f6
This byte transfer information is used to interactively update the progress bar as you might have noticed.

**Comment by ikwyl6** [comment]
Yes but I was referring to the transfer speeds.

**Comment by MWP** [comment]
Unless im mistaken, there appears to be no way of showing which files are actually being push/pulled?
Can this be added?

**Comment by odeke-em** [comment]
@MWP that is right. Currently printing that information would mess up the pretty print of the progress bar, but no worries that's why OSS forks work, I'll try to see a way of adding that plus @ikwyl6's request into the progress reporter.

**Comment by MWP** [comment]
Ok. Is there a log file anywhere? Any output to syslog?
Sorry, but it seems rather strange that there is no more detailed information anywhere about what drive is actually doing.
We all know how temperamental cloud storage can be, so not seeing any kind of information is rather nerve-racking.

**Comment by odeke-em** [comment]
Nope. I am more of the view that only give me information when things fail, imagine every recursive move on your terminal showed you actually output, I personally wouldn't like that. When errors are encountered, for sure, I personally would like to see output only then or after completion.
However, I understand what you mean there is ground for both expectations.

**Comment by odeke-em** [comment]
I'll probably add a verbosity mode or so. We'll see :)
